src/blowfish.js

```diff
@@ -0,0 +1,451 @@
+     *
+     * @param {Object} ctx The object has pbox and sbox.
+     * @param {Array} key An array of 32-bit words.
+     * @param {int} keysize The length of the key.
+     *
+     * @example
+     *
+     *     BlowFishInit(BLOWFISH_CTX, key, 128/32);
+     */
+    function BlowFishInit(ctx, key, keysize)
+    {
+        for(let Row = 0; Row < 4; Row++)
+        {
+            ctx.sbox[Row] = [];
+            for(let Col = 0; Col < 256; Col++)
+            {
+                ctx.sbox[Row][Col] = ORIG_S[Row][Col];
+            }
+        }
+
+        let keyIndex = 0;
+        for(let index = 0; index < N + 2; index++)
+        {
+            ctx.pbox[index] = ORIG_P[index] ^ key[keyIndex];
+            keyIndex++;
+            if(keyIndex >= keysize)
+            {
+                keyIndex = 0;
+            }
+        }
+
+        let Data1 = 0;
+        let Data2 = 0;
+        let res = 0;
+        for(let i = 0; i < N + 2; i += 2)
+        {
+            res = BlowFish_Encrypt(ctx, Data1, Data2);
+            Data1 = res.left;
+            Data2 = res.right;
+            ctx.pbox[i] = Data1;
+            ctx.pbox[i + 1] = Data2;
+        }
+        
+        for(let i = 0; i < 4; i++)
+        {
+            for(let j = 0; j < 256; j += 2)
+            {
+                res = BlowFish_Encrypt(ctx, Data1, Data2);
+                Data1 = res.left;
+                Data2 = res.right;
+                ctx.sbox[i][j] = Data1;
+                ctx.sbox[i][j + 1] = Data2;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * BF block cipher algorithm.
+     */
+    var BF = C_algo.BF = BlockCipher.extend({
+        _doReset: function () {
+            // Skip reset of nRounds has been set before and key did not change
+            if (this._keyPriorReset === this._key) {
+                return;
+            }
+
+            // Shortcuts
+            var key = this._keyPriorReset = this._key;
+            var keyWords = key.words;
+            var keySize = key.sigBytes / 4;
+
+            //Initialization pbox and sbox
+            BlowFishInit(BLOWFISH_CTX, keyWords, keySize);
+        },
+
+        encryptBlock: function (M, offset) {
+            var res = BlowFish_Encrypt(BLOWFISH_CTX, M[offset], M[offset + 1]);
+            M[offset] = res.left;
+            M[offset + 1] = res.right;
+        },
+
+        decryptBlock: function (M, offset) {
+            var res = BlowFish_Decrypt(BLOWFISH_CTX, M[offset], M[offset + 1]);
+            M[offset] = res.left;
+            M[offset + 1] = res.right;
+        },
+
+        blockSize: 64/32,
+
+        keySize: 128/32,
+
+        ivSize: 64/32
+    });
+
+    /**
+     * Shortcut functions to the cipher's object interface.
+     *
+     * @example
+     *
+     *     var ciphertext = CryptoJS.BF.encrypt(message, key, cfg);
+     *     var plaintext  = CryptoJS.BF.decrypt(ciphertext, key, cfg);
+     */
+    C.BF = BlockCipher._createHelper(BF);
```

Comment:
Would prefer to name it `C.Blowfish` instead of using abbreviation.

Reply:
Sounds good, Done!